ifrs17/Utils/Queries.ipynb

```diff
@@ -217,14 +217,26 @@
         {
             "cell_type": "code",
             "source": [
-                "public static async Task<Dictionary<string, Dictionary<int, YieldCurve>>> LoadCurrentYieldCurveAsync(this IQuerySource querySource, Args args, ",
-                "\n                                                                                                      IEnumerable<DataNodeData> dataNodes)",
-                "\n{",
-                "\n    var contractualCurrenciesInScope = dataNodes.Select(dn => dn.ContractualCurrency).ToHashSet();",
-                "\n    return (await querySource.LoadCurrentAndPreviousParameterAsync<YieldCurve>(args, ",
+                "public static async Task<Dictionary<string, Dictionary<int, YieldCurve>>> LoadCurrentYieldCurveAsync(this IQuerySource querySource, Args args,",
+                "\n                                                                                                                                IEnumerable<DataNodeData> dataNodes)",
+                "\n{    ",
+                "\n    var currentYieldCurveByGoc = new Dictionary<string,  Dictionary<int, YieldCurve>>();",
+                "\n    ",
+                "\n    foreach (var dn in dataNodes)",
+                "\n    {",
+                "\n        var loadedYc = await querySource.LoadCurrentAndPreviousParameterAsync<YieldCurve>(args, ",
                 "\n                                                                               x => x.Currency,",
-                "\n                                                                               x => contractualCurrenciesInScope.Contains(x.Currency)",
-                "\n                                                                               ));",
+                "\n                                                                               x => x.Currency == dn.ContractualCurrency ",
+                "\n                                                                                    && (dn.ValuationApproach == ValuationApproaches.VFA ",
+                "\n                                                                                                                ? x.Name == dn.YieldCurveName ",
+                "\n                                                                                                                : true));",
+                "\n        if(!loadedYc.TryGetValue(dn.ContractualCurrency, out var currentYcDict) || currentYcDict == null || currentYcDict?.Values == null)",
```

Comment:
These things should never happen --> any check on YC should be in the setter, not the getter.

Reply:
Should that check be done inside the task "LoadCurrentAndPreviousParameterAsync" ?

![image](https://user-images.githubusercontent.com/120557622/211319703-7b301667-91aa-4f07-a274-a2fce18f459c.png)

Reply:
@dtrzesniak Fixed with the last commit.